src/isar/state_machine/state_machine.py

```diff
@@ -257,11 +259,20 @@ def _mission_finished(self) -> None:
         partially_fail_statuses = fail_statuses + [TaskStatus.PartiallySuccessful]
 
         if all(task.status in fail_statuses for task in self.current_mission.tasks):
+            self.current_mission.user_message = UserMessage(
+                full_text=f"The mission failed because all tasks in the mission failed",
```

Comment:
Should we add the reasons from the failing tasks here (as done for steps in the task messages)?

Reply:
I'm assuming 'step' here means 'task'.
I agree that this would be nice functionality (Checking each task to see why it failed), and probably a good way to go.
Maybe we should add error codes to our error system and the StatusReason for mission can then say "Mission failed because all tasks failed (errorCode1, errorCode2 etc..)"